lib/components/popup/TopicPopup.dart

```diff
@@ -5,32 +5,38 @@ import 'package:off_top_mobile/components/recordingSession/recorder.dart';
 import 'package:shared_preferences/shared_preferences.dart';
 
 class MyTopicDialog extends StatefulWidget {
-  final ValueChanged<String> onTopicChanged;
+  const MyTopicDialog({Key key, this.onTopicChanged}) : super(key: key);
 
-  MyTopicDialog({Key key, this.onTopicChanged}): super(key: key);
+  final ValueChanged<String> onTopicChanged;
 
   @override
   _MyTopicDialogState createState() => _MyTopicDialogState();
 }
 
 class _MyTopicDialogState extends State<MyTopicDialog> {
-  final String title = "Select A Topic", buttonText = "Select";
-  final tileTitles = ['Computer Science', 'Sports', 'Food'];
-  List<bool> listCheck = [];
+  final String title = 'Select A Topic', buttonText = 'Select';
+  final List<String> tileTitles = <String>[
+    'Computer Science',
+    'Sports',
+    'Food'
+  ];
+  List<bool> listCheck = <bool>[];
 
   @override
   void initState() {
-    setState(() {
-      for (int i = 0; i < tileTitles.length; i++) {
-        listCheck.add(false);
-      }
-    });
+    setState(
+      () {
+        for (int i = 0; i < tileTitles.length; i++) {
+          listCheck.add(false);
+        }
+      },
+    );
     super.initState();
   }
 
-  void setSessionPreferences(String value) async {
-    SharedPreferences prefs = await SharedPreferences.getInstance();
-    prefs.setString("session", value);
+  static Future<bool> setSessionPreferences(String value) async {
```

Comment:
i don't feel comfortable with making this static. Is this part of the new flutter version?

Reply:
This one was more of a warning rather than a must do. If you feel uncomfortable, we can remove the static!